// retell-sdk/resources/chat-agent.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatAgent = void 0;
const resource_1 = require("../resource.js");
const core_1 = require("../core.js");
class ChatAgent extends resource_1.APIResource {
    /**
     * Create a new chat agent
     *
     * @example
     * ```ts
     * const chatAgentResponse = await client.chatAgent.create({
     *   response_engine: {
     *     llm_id: 'llm_234sdertfsdsfsdf',
     *     type: 'retell-llm',
     *   },
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/create-chat-agent', { body, ...options });
    }
    retrieve(agentId, query = {}, options) {
        if ((0, core_1.isRequestOptions)(query)) {
            return this.retrieve(agentId, {}, query);
        }
        return this._client.get(`/get-chat-agent/${agentId}`, { query, ...options });
    }
    /**
     * Update an existing chat agent
     *
     * @example
     * ```ts
     * const chatAgentResponse = await client.chatAgent.update(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    update(agentId, params, options) {
        const { version, ...body } = params;
        return this._client.patch(`/update-chat-agent/${agentId}`, { query: { version }, body, ...options });
    }
    list(query = {}, options) {
        if ((0, core_1.isRequestOptions)(query)) {
            return this.list({}, query);
        }
        return this._client.get('/list-chat-agents', { query, ...options });
    }
    /**
     * Delete an existing chat agent
     *
     * @example
     * ```ts
     * await client.chatAgent.delete(
     *   'oBeDLoLOeuAbiuaMFXRtDOLriTJ5tSxD',
     * );
     * ```
     */
    delete(agentId, options) {
        return this._client.delete(`/delete-chat-agent/${agentId}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
    /**
     * Get all versions of a chat agent
     *
     * @example
     * ```ts
     * const chatAgentResponses =
     *   await client.chatAgent.getVersions(
     *     '16b980523634a6dc504898cda492e939',
     *   );
     * ```
     */
    getVersions(agentId, options) {
        return this._client.get(`/get-chat-agent-versions/${agentId}`, options);
    }
    /**
     * Publish the latest version of the chat agent and create a new draft chat agent
     * with newer version.
     *
     * @example
     * ```ts
     * await client.chatAgent.publish(
     *   '16b980523634a6dc504898cda492e939',
     * );
     * ```
     */
    publish(agentId, options) {
        return this._client.post(`/publish-chat-agent/${agentId}`, {
            ...options,
            headers: { Accept: '*/*', ...options?.headers },
        });
    }
}
exports.ChatAgent = ChatAgent;
